tgui/packages/tgui/interfaces/PlumbingService.js

```diff
@@ -0,0 +1,114 @@
+import { classes } from 'common/react';
+import { Window } from '../layouts';
+import { useBackend, useLocalState } from '../backend';
+import { Section, Tabs, Button, Stack, Box } from '../components';
+import { ColorItem, LayerSelect } from './RapidPipeDispenser';
+
```

Comment:
```
type Data = {
    categories: Category[];
    icon: string;
};

type Category = {
  cat_name: string;
  recipes: Recipe[];
};

type Recipe = {
  index: number;
  selected: BooleanLike;
  name: string;
};
```

convert the file to .tsx and see if this works (don't hate me, I typed this out here).
Let me know if you need help changing the rest of the file over. 

Typescript is there to make life easier, all JS devs should be using it

Reply:
Don't worry I'll have vs code if it dosent work.  I'll do that